test/mocks/MockCalldataDecoder.sol

```diff
@@ -1,40 +1,38 @@
 // SPDX-License-Identifier: MIT
 pragma solidity ^0.8.24;
 
-import {PositionConfig} from "../../src/libraries/PositionConfig.sol";
 import {CalldataDecoder} from "../../src/libraries/CalldataDecoder.sol";
 import {IV4Router} from "../../src/interfaces/IV4Router.sol";
 import {Currency} from "@uniswap/v4-core/src/types/Currency.sol";
+import {PoolKey} from "@uniswap/v4-core/src/types/PoolKey.sol";
 
 // we need to use a mock contract to make the calls happen in calldata not memory
 contract MockCalldataDecoder {
     using CalldataDecoder for bytes;
 
+    struct MintParams {
+        PoolKey poolKey;
+        int24 tickLower;
+        int24 tickUpper;
+        uint256 liquidity;
+        uint128 amount0Max;
+        uint128 amount1Max;
+        address owner;
+        bytes hookData;
+    }
```

Comment:
this is struct-ified to avoid IR? if so, maybe worth adding a small comment why its struct-ified in tests and only for mint

Reply:
yes will do

Reply:
and yes to avoid ir lol